relay-server/src/utils/kafka.rs

```diff
@@ -0,0 +1,23 @@
+use rdkafka::producer::{DeliveryResult, ProducerContext};
+use rdkafka::ClientContext;
+
+use relay_common::{metric, LogError};
+
+use crate::metrics::RelayCounters;
+
+pub struct CaptureErrorContext;
+
+impl ClientContext for CaptureErrorContext {}
+
+impl ProducerContext for CaptureErrorContext {
+    type DeliveryOpaque = ();
+    fn delivery(&self, result: &DeliveryResult, _delivery_opaque: Self::DeliveryOpaque) {
+        if let Err((error, _message)) = result {
+            log::error!("callback producer error: {}", LogError(error));
```

Comment:
_nit_: Since this is logged in Sentry, maybe make this more verbose?

```suggestion
            log::error!("failed to produce message to kafka: {}", LogError(error));
```

Reply:
Makes sense, but I think we should include a hint that this callback was triggered asynchronously. What do you think about now?

Reply:
Ah, missed your question earlier. Yeah, looks good now!